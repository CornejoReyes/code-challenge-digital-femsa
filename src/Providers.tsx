import { NavigationContainer } from '@react-navigation/native';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { NativeBaseProvider } from 'native-base';
import { PropsWithChildren } from 'react';

const queryClient = new QueryClient();
const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export interface Props extends PropsWithChildren {}

export default function Providers({ children }: Props) {
  return (
    <NativeBaseProvider initialWindowMetrics={inset}>
      <QueryClientProvider client={queryClient}>
        <NavigationContainer>{children}</NavigationContainer>
      </QueryClientProvider>
    </NativeBaseProvider>
  );
}
