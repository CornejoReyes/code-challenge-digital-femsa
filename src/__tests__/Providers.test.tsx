import { render, screen } from '@testing-library/react-native';
import { Text } from 'native-base';
import Providers from '../Providers';

describe('<Providers />', () => {
  it('renders correctly', () => {
    render(
      <Providers>
        <Text>Children</Text>
      </Providers>,
    );
    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.getAllByText(/children/i)).toHaveLength(1);
  });
});
