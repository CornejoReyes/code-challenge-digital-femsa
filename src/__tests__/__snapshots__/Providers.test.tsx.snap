// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Providers /> renders correctly 1`] = `
<Text
  dataSet={{}}
  style={
    {
      "backgroundColor": undefined,
      "color": "#171717",
      "fontFamily": undefined,
      "fontSize": 14,
      "fontStyle": "normal",
      "fontWeight": "400",
      "letterSpacing": 0,
      "lineHeight": 21,
      "textDecorationLine": undefined,
    }
  }
>
  Children
</Text>
`;
